### DISCUSSION TITLE
MaskMap for Specific Irrigation Area

### DISCUSSION THREAD

**Discussion by Guo-icloud** [iiasa/CWatM#148] [Q&A]
Dear Professor,

Hi! I am currently working on learning the CWatM model and have encountered an issue. To focus more specifically on an irrigation area rather than the entire Yellow River basin, I replaced the coordinates in the MaskMap with a .tif file specific to my area of interest. The entry in the settings file is as follows:
MaskMap = $(FILE_PATHS:PathRoot)/areamaps/hetao_WGS1984_30min.tif

This .tif file has been saved in the areamaps folder. While the "Gauges" is setting as follows: 
1. It still use coordinates from the Yellow River. OR
2. It is replaced with the .tif file. 
Both of the settings, I encountered the following error message and am unsure how to proceed.
![Error when CWatM settings for specific irrigation area_20240715 pic](https://github.com/user-attachments/assets/184f79b1-3670-4e46-a758-680c89aac1c5)

And while the study area is an irrigation area, I wonder if including the upstream basin above this irrigation area in the simulation might provide better insights into the hydrological processes affecting it.

Could you please advise on how I might resolve this issue? Any suggestions or guidance would be greatly appreciated. Thank you. :)

Best wishes

**Comment by dof1985** [comment]
Hi @Guo-icloud 

I can't reproduce your problem, so it is hard to assist you. Here are some few hints to go forward.

The error is triggered during the initial setup of the model; specifically when try to load the CellArea input. Have you made any changes to this file/setting?
It further suggests that when it is trying to compress the cell area map that when it is trying to compress the cell area map, and checking of nan values - it receives an empty array. 

I would propose that you focus on data_handling.py --> loadmap function. Go to the code and try to track the dimensions of mapnp (e.g., by print(mapnp.shape)); At some point it would probably becomes an empty array.

Please share your findings.
Dor
